File: app/src/main/java/com/kongpf8848/dmail/util/TokenUtils.java
package com.kongpf8848.dmail.util;

import android.util.Base64;

import org.json.JSONObject;

import java.nio.charset.StandardCharsets;

public class TokenUtils {

    public static String getEmailFromIdToken(String idToken) {
        try {
            String[] split = idToken.split("\\.");
            String jwtBody = split[1];
            byte[] decodedBodyBytes = Base64.decode(jwtBody, Base64.URL_SAFE);
            String body = new String(decodedBodyBytes, StandardCharsets.UTF_8);
            JSONObject jsonObject=new JSONObject(body);
            return jsonObject.optString("email");
        } catch (Exception e) {
            e.printStackTrace();
        }
        return "";
    }
}

File: app/src/main/java/com/kongpf8848/dmail/util/TokenUtils.kt
package com.kongpf8848.dmail.util

import android.util.Base64
import org.json.JSONObject
import java.lang.Exception
import java.nio.charset.StandardCharsets

object TokenUtils {
    fun getEmailFromIdToken(idToken: String): String {
        try {
            val split = idToken.split("\\.").toTypedArray()
            val jwtBody = split[1]
            val decodedBodyBytes = Base64.decode(jwtBody, Base64.URL_SAFE)
            val body = String(decodedBodyBytes, StandardCharsets.UTF_8)
            val jsonObject = JSONObject(body)
            return jsonObject.optString("email")
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return ""
    }
}